aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -822,6 +822,7 @@ Tensor index_select_mps(const Tensor& self, int64_t dim, const Tensor& index) {
     runMPSGraph(stream, cachedGraph->graph(), feeds, results);
   }
   namedinference::propagate_names_if_nonempty(self, maybe_outnames);
+  restride_contiguous_(self);
```

Comment:
I don't think this is ok? What if self is a smaller view of a larger Tensor like:
```
t = torch.rand(100, 100, 100)
self = t.select(0, 0).select(-1, 0)
```
The restriding that you do here will make self read completely unrelated memory within t.

Reply:
With the example you provided, the transformation of the size and stride is essentially

size `(100, 100, 100)` -> `(100,)` 
stride `(10000, 100, 1)` -> `(100,)`.

The stride after `restride_contiguous_` would be `(1,)` .

In MPS, the internal data storage is always contiguous (no stride concept, that's why we use `gatheViewTensor` trick to gather the cached contiguous data when the incoming tensor is a view.) and we access the data of a tensor by accessing the pointer of the `MTLBuffer` first and then reading its data contiguously. In this in-place case, we just write the output to the first `100` element size of the input buffer and don't care about the rest, and at this point the stride is wrong so we reset it. Of course in conjunction with non-zero `storage_offset` we may need to do some special care of that.

In out-of-the-place cases, things become simpler as the storage of input tensor and output tensor is different. (In pure view operations the storage is the same, but the computation wouldn't actually flow into MPS Graph kernel because they're just stride & size rearrangement)

Lastly if the view's size is larger than the storage, it's considered invalid in the first place, and we don't need to handle it.

Please let me know if I missed something :)

Reply:
> The stride after restride_contiguous_ would be (1,) .

But as you mention the storage offset is completely wrong. So your op will have written at the wrong place in the big Tensor!

> Lastly if the view's size is larger than the storage, it's considered invalid in the first place, and we don't need to handle it.

Technically this is allowed when calling as_strided, but that's fine to just error out here.

Reply:
Do you think if we can simply write the buffer from the beginning and reset the storage offset to 0? Will it violate the PyTorch's philosophy or something?

Reply:
The problem is that this part of the storage might be used by other Tensors and you should not override their content.